src/Chapter4.hs

```diff
@@ -628,7 +661,7 @@ Can you implement a monad version of AND, polymorphic over any monad?
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+andM mx my = mx >>= \x -> my >>= \y -> pure $ x && y
```

Comment:
Regarding your question, in the test case `Just False` is there on purpose.

The ordinary `&&` if lazy on the second argument. Meaning that if it meets `False` as the first argument, there is no need to calculate another arg. In `andM` we would like to keep the same behaviour, so the

```
Just False `andM` _
```

should be `Just False` :slightly_smiling_face: 


If you don't use the `>>=` operator, then the `Monad` constraint is too strong, and it is enough to specify only `Applicative`. However, with `Applicative` you are not able to implement a lazier version of `andM`.

Reply:
For my code this test fails. How can I change code to pass this test case? Don't understand yet.

Reply:
Sure, no problem! All you need to do is to first check on the first value and stop lazily in case it is `False`. The code could look. like this:

```haskell
andM ma mb = ma >>= \a -> if a then mb else pure False
```

Reply:
I see. Thank you for your help and review :)

Could you suggest what can be done next to improve Haskell skills?